x-pack/plugins/cloud/public/plugin.tsx

```diff
@@ -118,7 +120,12 @@ export class CloudPlugin implements Plugin<CloudSetup> {
       base_url: baseUrl,
     } = this.config;
 
-    this.isCloudEnabled = getIsCloudEnabled(id);
+    if (id) {
+      // We use the Hashed Cloud Deployment ID as the userId in the Cloud Experiments
+      cloudExperiments?.identifyUser(sha256(id), {
+        kibanaVersion: this.initializerContext.env.packageInfo.version,
+      });
```

Comment:
TODO: Extend the user metadata to include the boolean `inTrial` and `isOwnerElasticStaff`. Because of amount of changes required to retrieve that, I'll implement that in a separate PR.